compel/arch/x86/src/lib/infect.c

```diff
@@ -415,8 +415,13 @@ int compel_get_task_regs(pid_t pid, user_regs_struct_t *regs, user_fpregs_struct
 			set_user_reg(regs, ip, get_user_reg(regs, ip) - 2);
 			break;
 		case -ERESTART_RESTARTBLOCK:
-			pr_warn("Will restore %d with interrupted system call\n", pid);
-			set_user_reg(regs, ax, -EINTR);
+			if (pre_dump) {
```

Comment:
Would it make sense to use `ERESTART_RESTARTBLOCK` also when `criu dump` is used with `--leave-running`?

Reply:
Yes, I think --leave-running should do the same. I have made some changes to the patch.
```
parasite_ctl = parasite_infect_seized(pid, item, &vmas, opts.final_state == TASK_ALIVE);
```